### PR TITLE
fix: allow search on fields number

### PR DESCRIPTION
**Pull request by arnaud-moncel**
## Pull Request checklist:

- [X] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [X] Create automatic tests
- [X] Test manually the implemented changes
- [X] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by arnaudbesnier** [comment]
Task linked: [CU-80q41v P3 - Search with field number not working](https://app.clickup.com/t/80q41v)

File: src/services/search-builder.js [hunk 1]
```diff
@@ -147,6 +147,12 @@ function SearchBuilder(model, opts, params, fieldNamesRequested) {
           );
           pushCondition(condition, columnName);
         }
+      } else if (field.type === 'Number') {
+        const value = Number(params.search);
+        if (value) {
+          condition[field.field] = value;
+          pushCondition(condition, field.field);
+        }
```
Comment (slimee) [inline src/services/search-builder.js:155]: It seems searching 0 will return no results. 
Using isNaN may help.

**Review by slimee — CHANGES_REQUESTED** [review]
yeah nice shot captain :) 

I've add a small point, please look.

**Review by slimee — APPROVED** [review]
🚀

**Pull request merged**